specs/SensitivityLabel.md

```diff
@@ -0,0 +1,541 @@
+/// whether sensitivity labels have been detected, are being processed,
+/// or are fully determined for the current web page content.
+[v1_enum]
+typedef enum COREWEBVIEW2_SENSITIVITY_LABEL_STATE {
+  /// Indicates that none of the loaded pages are in the allow list. Hence 
+  /// none will report sensitivity labels. 
+  COREWEBVIEW2_SENSITIVITY_LABEL_STATE_NONE,
+  /// Indicates that WebView2 has loaded pages from the allow list that can
+  /// report sensitivity labels, but the label determination is not yet 
+  /// complete.
+  COREWEBVIEW2_SENSITIVITY_LABEL_STATE_UNDETERMINED,
+  /// Indicates that WebView2 has loaded pages from the allow list,
+  /// and those pages have provided label information.
+  COREWEBVIEW2_SENSITIVITY_LABEL_STATE_DETERMINED,
+} COREWEBVIEW2_SENSITIVITY_LABEL_STATE;
+
+/// Represents the kind of sensitivity label applied to web page.
+/// Sensitivity labels are used to classify and protect content based on
+/// its sensitivity level.
+[v1_enum]
+typedef enum COREWEBVIEW2_SENSITIVITY_LABEL_KIND {
+  /// Represents an unknown or unsupported sensitivity label.
+  COREWEBVIEW2_SENSITIVITY_LABEL_KIND_UNKNOWN,
+  /// Represents a Microsoft Information Protection (MIP) sensitivity label.
+  COREWEBVIEW2_SENSITIVITY_LABEL_KIND_MIP,
+} COREWEBVIEW2_SENSITIVITY_LABEL_KIND;
+
+/// Interface for different sensitivity label kinds used in WebView2.
+/// This interface provides functionality for accessing sensitivity
+/// label information applied to web page. Different label types
+/// (such as Microsoft Information Protection labels) provide
+/// specific label information and metadata.
+[uuid(5c27e6f2-baa6-5646-b726-db80a77b7345), object, pointer_default(unique)]
+interface ICoreWebView2SensitivityLabel : IUnknown {
+  /// Gets the type of the sensitivity label applied to the web page.
+  /// This property identifies which sensitivity label system is being used
+  /// (such as Microsoft Information Protection or other label providers).
+  /// Applications can use this information to determine how to interpret
+  /// and handle the label data, as different label types may have different
+  /// metadata formats, protection requirements, and policy enforcement
+  /// mechanisms.
+  [propget] HRESULT LabelKind(
+      [out, retval] COREWEBVIEW2_SENSITIVITY_LABEL_KIND* value);
+}
+
+/// Interface for Microsoft Information Protection (MIP) sensitivity labels.
+/// This interface provides specific information about MIP labels, including
+/// label identification and organizational context.
+[uuid(1a562888-3031-5375-b8c5-8afd573e79c8), object, pointer_default(unique)]
+interface ICoreWebView2MipSensitivityLabel : IUnknown {
+  /// The unique identifier for the Microsoft Information Protection label.
+  /// This string contains a GUID that uniquely identifies the specific
+  /// sensitivity label within the organization's MIP policy configuration.
+  /// The GUID is of type GUIDv4 and follows the format:
```

Comment:
I don't think v4 is important. Indeed, MIP might decide to move past v4 and use v5 (namespace-based).